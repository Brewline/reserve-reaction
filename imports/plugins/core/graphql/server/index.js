import { Meteor } from "meteor/meteor";
import { WebApp } from "meteor/webapp";
import createApolloServer from "./createApolloServer";
import { baseContext } from "./buildMeteorContext";

const server = createApolloServer({
  context: baseContext,
  // XXX Eventually these should be from individual env variables instead
  debug: Meteor.isDevelopment,
  graphiql: Meteor.isDevelopment
});

// bind the specified paths to the Express server running Apollo + GraphiQL
WebApp.connectHandlers.use(server);
