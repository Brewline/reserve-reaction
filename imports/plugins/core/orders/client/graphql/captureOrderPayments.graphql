captureOrderPayments($input: CaptureOrderPaymentsInput!) {
  captureOrderPayments(input: $input) {
    order {
      _id
      payments {
        _id
        captureErrorCode
        captureErrorMessage
        status
      }
    }
  }
}
