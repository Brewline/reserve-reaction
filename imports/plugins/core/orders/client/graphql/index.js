import simpleGraphQLClient from "/imports/plugins/core/graphql/lib/helpers/simpleClient";
import getOpaqueIds from "/imports/plugins/core/core/client/util/getOpaqueIds";
import captureOrderPaymentsMutation from "./captureOrderPayments.graphql";
import updateOrderFulfillmentGroupMutation from "./updateOrderFulfillmentGroup.graphql";

const captureOrderPaymentsMutate = simpleGraphQLClient.createMutationFunction(captureOrderPaymentsMutation);
const updateOrderFulfillmentGroupMutate = simpleGraphQLClient.createMutationFunction(updateOrderFulfillmentGroupMutation);

export const captureOrderPayments = async ({ orderId, paymentIds, shopId }) => {
  // Convert MongoDB IDs to GraphQL IDs
  const conversionRequests = [
    { namespace: "Order", id: orderId },
    { namespace: "Shop", id: shopId }
  ];

  paymentIds.forEach((paymentId) => {
    conversionRequests.push({ namespace: "Payment", id: paymentId });
  });

  const [
    opaqueOrderId,
    opaqueShopId,
    ...opaquePaymentIds
  ] = await getOpaqueIds(conversionRequests);

  return captureOrderPaymentsMutate({
    input: {
      orderId: opaqueOrderId,
      paymentIds: opaquePaymentIds,
      shopId: opaqueShopId
    }
  });
};

export const updateOrderFulfillmentGroup = async ({
  orderId,
  orderFulfillmentGroupId,
  status,
  tracking
}) => {
  // Convert MongoDB IDs to GraphQL IDs
  const conversionRequests = [
    { namespace: "Order", id: orderId },
    { namespace: "OrderFulfillmentGroup", id: orderFulfillmentGroupId }
  ];

  const [
    opaqueOrderId,
    opaqueOrderFulfillmentGroupId
  ] = await getOpaqueIds(conversionRequests);

  return updateOrderFulfillmentGroupMutate({
    orderFulfillmentGroupId: opaqueOrderFulfillmentGroupId,
    orderId: opaqueOrderId,
    status,
    tracking
  });
};
