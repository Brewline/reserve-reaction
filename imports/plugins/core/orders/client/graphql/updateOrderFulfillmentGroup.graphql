updateOrderFulfillmentGroup($orderFulfillmentGroupId: ID!, $orderId: ID!, $status: String, $tracking: String) {
  updateOrderFulfillmentGroup(input: {
    orderFulfillmentGroupId: $orderFulfillmentGroupId,
    orderId: $orderId,
    status: $status,
    tracking: $tracking
  }) {
    order {
      fulfillmentGroups {
        status
        tracking
      }
    }
  }
}
