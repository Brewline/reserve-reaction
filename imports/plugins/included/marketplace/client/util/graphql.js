import simpleGraphQLClient from "/imports/plugins/core/graphql/lib/helpers/simpleClient";
import placeMarketplaceOrderWithStripeCardPaymentMutation from "./placeMarketplaceOrderWithStripeCardPayment.graphql";

export const placeMarketplaceOrderWithStripeCardPayment = simpleGraphQLClient.createMutationFunction(placeMarketplaceOrderWithStripeCardPaymentMutation);
