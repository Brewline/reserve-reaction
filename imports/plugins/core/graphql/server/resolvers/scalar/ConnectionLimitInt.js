import { GraphQLScalarType } from "graphql";
import { Kind } from "graphql/language";

const MAX_LIMIT = 50;

/**
 * Adjusts value to be between 1 and MAX_LIMIT, inclusive
 * @private
 */
function parseValue(value) {
  // Note that we do not have to do isNaN(value) check here because GraphQLScalarType will not call this for isNaN.
  // Instead it automatically changes isNaN to undefined.
  if (typeof value !== "number") return MAX_LIMIT;
  return Math.min(Math.max(1, value), MAX_LIMIT);
}

const description = `
An integer between 1 and 50, inclusive. Values less than 1 become 1 and
values greater than 50 become 50.
`;

export default new GraphQLScalarType({
  description,
  name: "ConnectionLimitInt",
  serialize: (value) => value,
  parseLiteral(ast) {
    if (ast.kind !== Kind.INT) return MAX_LIMIT;
    return parseValue(parseInt(ast.value, 10));
  },
  parseValue
});
