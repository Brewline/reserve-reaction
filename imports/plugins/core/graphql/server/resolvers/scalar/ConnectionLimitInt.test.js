import { Kind } from "graphql/language";
import ConnectionLimitInt from "./ConnectionLimitInt";

test("has a description", () => {
  expect(ConnectionLimitInt.description).toMatchSnapshot();
});

describe("serialization", () => {
  test("returns whatever is given", () => {
    expect(ConnectionLimitInt.serialize(undefined)).toBe(undefined);
    expect(ConnectionLimitInt.serialize(null)).toBe(null);
    expect(ConnectionLimitInt.serialize(1)).toBe(1);
    expect(ConnectionLimitInt.serialize(100)).toBe(100);
  });
});

describe("value parsing", () => {
  test("returns what is given if it's a number between 1 and 50, inclusive, or undefined", () => {
    expect(ConnectionLimitInt.parseValue(1)).toBe(1);
    expect(ConnectionLimitInt.parseValue(2)).toBe(2);
    expect(ConnectionLimitInt.parseValue(10)).toBe(10);
    expect(ConnectionLimitInt.parseValue(50)).toBe(50);
    expect(ConnectionLimitInt.parseValue(undefined)).toBe(undefined);
  });

  test("returns 50 if over 50", () => {
    expect(ConnectionLimitInt.parseValue(null)).toBe(50);
    expect(ConnectionLimitInt.parseValue(51)).toBe(50);
    expect(ConnectionLimitInt.parseValue(100)).toBe(50);
  });

  test("returns 1 if under 1", () => {
    expect(ConnectionLimitInt.parseValue(0)).toBe(1);
    expect(ConnectionLimitInt.parseValue(-10)).toBe(1);
  });

  test("returns undefined for isNaN", () => {
    expect(ConnectionLimitInt.parseValue(parseInt("f", 10))).toBe(undefined);
  });
});

const getLiteralForInt = (value) => ({ kind: Kind.INT, value });

describe("literal parsing", () => {
  test("returns what is given if it's a number between 1 and 50, inclusive", () => {
    expect(ConnectionLimitInt.parseLiteral(getLiteralForInt(1))).toBe(1);
    expect(ConnectionLimitInt.parseLiteral(getLiteralForInt(2))).toBe(2);
    expect(ConnectionLimitInt.parseLiteral(getLiteralForInt(10))).toBe(10);
    expect(ConnectionLimitInt.parseLiteral(getLiteralForInt(50))).toBe(50);
  });

  test("returns 50 if over 50", () => {
    expect(ConnectionLimitInt.parseLiteral(getLiteralForInt(51))).toBe(50);
    expect(ConnectionLimitInt.parseLiteral(getLiteralForInt(100))).toBe(50);
  });

  test("returns 1 if under 1", () => {
    expect(ConnectionLimitInt.parseLiteral(getLiteralForInt(0))).toBe(1);
    expect(ConnectionLimitInt.parseLiteral(getLiteralForInt(-10))).toBe(1);
  });
});
