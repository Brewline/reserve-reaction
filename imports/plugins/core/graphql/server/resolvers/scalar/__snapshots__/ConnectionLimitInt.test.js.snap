// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a description 1`] = `
"
An integer between 1 and 50, inclusive. Values less than 1 become 1 and
values greater than 50 become 50.
"
`;
